runtime/src/iree/hal/drivers/hip/hip_multi_queue_command_buffer.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_HAL_DRIVERS_HIP_MULTI_QUEUE_COMMAND_BUFFER_H__
+#define IREE_HAL_DRIVERS_HIP_MULTI_QUEUE_COMMAND_BUFFER_H__
+
+#include "iree/base/api.h"
+#include "iree/hal/command_buffer.h"
+#include "iree/hal/drivers/hip/dynamic_symbols.h"
+#include "iree/hal/drivers/hip/per_device_information.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+typedef struct iree_arena_block_pool_t iree_arena_block_pool_t;
+
+//===----------------------------------------------------------------------===//
+// iree_hal_command_buffer_t deferred record/replay wrapper
+//===----------------------------------------------------------------------===//
+
+// Records an command buffer that records into multiple command buffers
+// at a time based on the given queue affinity.
+//
+// After recording the underlying command buffers can be retrieved with
+// iree_hal_hip_multi_queue_command_buffer_get for submission.
+
+IREE_API_EXPORT iree_status_t iree_hal_hip_multi_queue_command_buffer_create(
+    iree_allocator_t host_allocator, uint32_t command_buffer_count,
```

Comment:
```suggestion
    iree_allocator_t host_allocator, iree_host_size_t command_buffer_count,
```
(mentioned elsewhere, iree_host_size_t for all counts/sizes/indices/etc unless required for a specific purpose)